### PR TITLE
generalize Boomerang for not isotropic reference measures

### PR DESCRIPTION
**Pull request by mschauer**

### CODE REVIEW

File: src/dynamics.jl [hunk 1]
```diff
@@ -78,12 +78,23 @@ function move_forward(τ, t, x, θ, B::Boomerang1d)
...
 """
-    reflect!(∇ϕx, θ, F::BouncyParticle, Boomerang)
+    reflect!(∇ϕx, x, θ, F::Boomerang)
 
 Reflection rule of sampler `F` at reflection time.
 x`: position, `θ`: velocity
 """
-function reflect!(∇ϕx, x, θ, ::Union{BouncyParticle, Boomerang})
+function reflect!(∇ϕx, x, θ, F::Boomerang)
+    θ .-= (2*dot(∇ϕx, θ)/normsq(F.Σ12*∇ϕx))*F.Σ*∇ϕx
```
Comment (mschauer) [inline src/dynamics.jl:87]:
```suggestion
    θ .-= (2*dot(∇ϕx, θ)/normsq(F.L*∇ϕx))*F.Σ*∇ϕx
```
Shouldn't that be the same?

File: src/types.jl [hunk 2]
```diff
@@ -41,15 +41,34 @@ BouncyParticle(Γ, μ, λ; ρ=0.0) = BouncyParticle(Γ, μ, λ, ρ)
     Boomerang(μ, λ) <: ContinuousDynamics
 
 Dynamics preserving the `N(μ, Σ)` measure (Boomerang)
-with refreshment time `λ`
+with refreshment time `λref`. `Γ` defines the sparse dependency graph of the target measure,
```
Comment (mschauer) [inline src/types.jl:44]: Shouldn't `Γ = inv(Σ)` here?

File: src/dynamics.jl (hunk 1 above)
Comment (SebaGraz) [reply src/dynamics.jl:87]: Yes! Then we don't need Σ12 anymore.

File: src/types.jl (hunk 2 above)
Comment (SebaGraz) [reply src/types.jl:44]: Right! I ll update the file today

Comment (SebaGraz) [reply src/types.jl:44]: Actually is that right?  Γ should be the precision matrix (or the dependency graph) of the target distribution while inv(Σ) should be the precision matrix of the reference measure. Right?

Comment (mschauer) [reply src/types.jl:44]: We need a dependency graph for the target, a precision for the reference measure and upper bounds. In part of the code we assume that all of this is the same. But the sparsity of the target is only used in the "fully local" algorithm. This needs to be cleaned up a bit

**Pull request closed without merging**